MetaMask/metamask-extension
# feat: display full terms of service during onboarding

## **Description**



During the onboarding process, the user is currently presented with a simple checkbox to agree to terms:

![Screenshot 2024-08-12 at 14 15 34](https://github.com/user-attachments/assets/ea351329-225a-4d8e-b9ad-d22888e8d2f6)

Legal would prefer we show the full terms of service, requiring the user to scroll the entire terms and agree at the end.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/29392?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3013

## **Manual testing steps**

1. Start onboarding
2. See checkbox but without full terms.
3. Verify full terms are displayed, requiring user to scroll and agree.

## **Screenshots/Recordings**



### **Before**



![Screenshot 2024-08-12 at 14 15 34](https://github.com/user-attachments/assets/ea351329-225a-4d8e-b9ad-d22888e8d2f6)

### **After**

https://github.com/user-attachments/assets/67014e57-c953-481c-bd27-70ea95165e48


_No response_

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
